### PR TITLE
Enable ComWrappers for NativeAOT on non-Windows

### LINKED ISSUE #76005: [NativeAot] ComWrappers don't work on Linux
```
[FAIL] System.Runtime.InteropServices.Tests.ReleaseTests.Release_ValidPointer_Success
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.ReleaseTests.Release_ValidPointer_Success() + 0x48
   at System.Runtime.InteropServices!<BaseAddress>+0xfda184
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0xd8
[FAIL] System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_ValidInterface_Success(o: Object { }, iidString: "00000000-0000-0000-C000-000000000046")
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_ValidInterface_Success(Object o, String iidString) + 0x48
   at System.Runtime.InteropServices!<BaseAddress>+0xff910c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
[FAIL] System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_ValidInterface_Success(o: Object { }, iidString: "1F906666-B388-4729-B78C-826BC5FD4245")
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_ValidInterface_Success(Object o, String iidString) + 0x48
   at System.Runtime.InteropServices!<BaseAddress>+0xff910c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
[FAIL] System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_NoSuchInterface_Success(o: Object { }, iidString: "00000000-0000-0000-0000-000000000000")
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_NoSuchInterface_Success(Object o, String iidString) + 0x44
   at System.Runtime.InteropServices!<BaseAddress>+0xff910c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
[FAIL] System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_NoSuchInterface_Success(o: Object { }, iidString: "639610C8-26EA-48BB-BD74-4BC18EECC6EA")
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.QueryInterfaceTests.QueryInterface_NoSuchInterface_Success(Object o, String iidString) + 0x44
   at System.Runtime.InteropServices!<BaseAddress>+0xff910c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
[FAIL] System.Runtime.InteropServices.Tests.AddRefTests.AddRef_ValidPointer_Success
System.PlatformNotSupportedException : Operation is not supported on this platform.
   at System.Runtime.InteropServices.ComWrappers.GetOrCreateComInterfaceForObject(Object instance, CreateComInterfaceFlags flags) + 0x28
   at System.Runtime.InteropServices.Tests.AddRefTests.AddRef_ValidPointer_Success() + 0x48
   at System.Runtime.InteropServices!<BaseAddress>+0xfda184
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0xd8
```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #76005

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/tests/Interop/COM/ComWrappers/API/Program.cs [hunk 1]
```diff
@@ -713,8 +714,12 @@ static int Main()
                 ValidateBadComWrapperImpl();
                 ValidateRuntimeTrackerScenario();
                 ValidateQueryInterfaceAfterManagedObjectCollected();
-                ValidateAggregationWithComObject();
-                ValidateAggregationWithReferenceTrackerObject();
+
+                if (!PlatformDetection.IsNativeAot)
```
Comment (jkotas) [inline src/tests/Interop/COM/ComWrappers/API/Program.cs:718]: Are these known issue?

Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/API/Program.cs:718]: Yes, this is tracked by #74620

File: src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj [hunk 2]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <NativeAotIncompatible>false</NativeAotIncompatible>
```
Comment (MichalStrehovsky) [inline src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: Why is this needed? I would not expect us to have to enable ComWrappers tests one-by-one (it runs the risk of new ComWrappers tests not running with nativeaot by accident.)

Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: ComWrappers on NativeAOT only supports the basic API scenarios, so some of the other tests require features that have yet to be implemented. Also some of the tests require app manifests that I don't believe work with the existing binplace mechanism for the NativeAOT runtime tests.

Comment (MichalStrehovsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: What I have trouble with is finding the spot that sets this to true for this project - why do we need to set it to false?

Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: It's set to true in Directory.Build.props in Interop/COM.

Comment (MichalStrehovsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: That one is conditioned on IsManagedCOMClient so I would not expect it to apply

Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: I'll try to figure out why these tests weren't being run otherwise.

Comment (MichalStrehovsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: I think they build - NativeAotIncompatible prevents things from building but here's a log from #84404 https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/237437/logs/1914 that has:

```
2023-04-13T02:33:59.7181103Z D:\a\_work\1\s\artifacts\TargetingPack\xunit.assert.dll : warning IL2104: Assembly 'xunit.assert' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [D:\a\_work\1\s\src\tests\Interop\COM\ComWrappers\API\ComWrappersTests.csproj] [D:\a\_work\1\s\src\tests\build.proj]
```

so we do look at it.

Comment (jkoritzinsky) [reply src/tests/Interop/COM/ComWrappers/API/ComWrappersTests.csproj:5]: Looks like issues.targets has the exclusion. I'll remove it from there.
